src/SignhostAPIClient/Rest/SignhostApiReceiver.cs

```diff
@@ -35,36 +35,63 @@ public SignhostApiReceiver(SignhostApiReceiverSettings receiverSettings)
 			out Transaction postbackTransaction)
 		{
 			postbackTransaction = null;
-			string[] postbackChecksumArray;
 			string postbackChecksum;
 			string calculatedChecksum;
 			PostbackTransaction postback;
 
-			postback = JsonConvert.DeserializeObject<PostbackTransaction>(body);
-			postbackTransaction = postback;
+			postback = DeserializeToPostbackTransaction(body);
+			postbackChecksum = GetChecksumFromHeadersOrPostback(headers, postback);
+			bool parametersAreValid = ChecksumCalculationParametersAreValid(postbackChecksum, postback);
 
-			if (headers.TryGetValue("Checksum", out postbackChecksumArray)) {
-				postbackChecksum = postbackChecksumArray.First();
-			} else {
-				postbackChecksum = postback.Checksum;
-			}
-
-			if (!string.IsNullOrEmpty(postbackTransaction.Id) &&
-				!string.IsNullOrEmpty(postbackChecksum))
-			{
-				using (var sha1 = SHA1.Create()) {
-					var preCalculatedChecksum = sha1.ComputeHash(Encoding.UTF8.GetBytes(
-						$"{postback.Id}||{(int)postback.Status}|{settings.SharedSecret}"));
-					calculatedChecksum = BitConverter.ToString(
-					preCalculatedChecksum)
-						 .Replace("-", string.Empty)
-						.ToLower();
-				}
+			if (parametersAreValid) {
+				calculatedChecksum = CalculateChecksumFromPostback(postback);
+				postbackTransaction = postback;
 			} else {
 				return false;
 			}
 
 			return Equals(calculatedChecksum, postbackChecksum);
 		}
+
+		private string CalculateChecksumFromPostback(PostbackTransaction postback)
+		{
+			string calculatedChecksum;
+			using (var sha1 = SHA1.Create()) {
+				var checksumBytes = sha1.ComputeHash(Encoding.UTF8.GetBytes(
+					$"{postback.Id}||{(int)postback.Status}|{settings.SharedSecret}"));
+				return calculatedChecksum = BitConverter.ToString(checksumBytes)
+					.Replace("-", string.Empty)
+					.ToLower();
+			}
+		}
+
+		private PostbackTransaction DeserializeToPostbackTransaction(string body)
+		{
+			return JsonConvert.DeserializeObject<PostbackTransaction>(body);
+		}
+
+		private string GetChecksumFromHeadersOrPostback(
+			IDictionary<string, string[]> headers,
+			PostbackTransaction postback)
+		{
+			string[] postbackChecksumArray;
+			string postbackChecksum;
+			if (headers.TryGetValue("Checksum", out postbackChecksumArray)) {
+				return postbackChecksumArray.First();
+			}
+			else {
+				return postbackChecksum = postback.Checksum;
+			}
+		}
+
+		private bool ChecksumCalculationParametersAreValid(string postbackChecksum, PostbackTransaction postback)
+		{
+			if (!string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id)) {
+				return true;
+			}
+			else {
+				return false;
+			}
```

Comment:
This one can retry the result immediatly there is no need for a boolean function to check if something is true to return true instead you can just return the check immediatly eg:
`return string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id);`

and if required you can shorten the signature further to:
```
private bool ChecksumCalculationParametersAreValid(string postbackChecksum, PostbackTransaction postback)
		=> !string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id);
```


ps. I find the naming of `ChecksumCalculationParametersAreValid` confusing as I am expecting method parameters / arguments to be validated (which I might not want as an external call), however instead it appears to be something like HasValidChecksumProperties (still not the best name though).

Reply:
I'll implement your suggestion to complete the commit. 
While speculating, I came across this 'Guard Clauses', which might be an interesting alternative:
[https://www.nuget.org/packages/Ardalis.GuardClauses](url)